doc/source/contributor/ofed.rst

```diff
@@ -25,31 +23,72 @@ Before building OFED packages, the workflow will ensure that:
-the kernels we provide in release train are unsupported by OFED. To accomplish this we
-will need to use the doca-kernel-support from the doca-extra repository.
+Currently we only support building Rocky Linux 9 OFED kernel module packages.
 
-We will need to instll dependencies in order to build the OFED kernel modules, and these
-are installed at the beginning of the build playbook. We also install base and appstream
-dependencies of userspace OFED packages here, this is intended to stop these dependencies
-being pulled in later when we download the OFED packages from the doca-host repository.
+The Build OFED module workflow will check that the filesystem is configured (noexec disabled)
+to allow the DOCA build script to run. The workflow will also install any necessary dependencies
+for the module build.
 
-At the end of the playbook following the kernel module build, the OFED userspace packages
-are downloaded from the upstream repository in order to upload these to Ark.
+The build script will output a ``doca-kernel-repo`` RPM which contains all kernel modules built
+as part of the workflow. When this RPM is installed, the repofile is created pointing to the
+modules in ``/usr/share/doca-host-<doca-version>/Modules/<kernel-version>/`` on the host.
 
 push-ofed
 ---------
 
-As we're not syncing OFED from any upstream source, and are instead creating our own
-repository of custom packages, we will be required to setup the Pulp distribution/publication
-and upload the content directly to Ark. This playbook uses the Pulp CLI to upload the RPMs
-to Ark.
+As mentioned above, the DOCA repository is synced into the ``doca`` repository in Ark. This workflow
+will upload the ``doca-kernel-repo`` RPM to a separate repository named ``doca-modules``. The version
+for this repository is set in ``pulp-repo-versions.yml`` and is disabled for local pulp syncs by
+default.
+
+Install process
+===============
+
+Release Train configuration
+---------------------------
+
+DOCA repositories will need to be synced to the local Pulp service. This can be enabled
+in ``ofed.yml``:
+
+.. code-block:: yaml
+
+  stackhpc_pulp_sync_ofed_modules: true
+
+With kernel module syncing enabled, the local Pulp can be synced with Ark by running:
+
+.. code-block:: console
+
+  kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-sync.yml
+  kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-publish.yml
+
+DOCA repositories can be templated to hosts by running Kayobe host configure.
+
+.. code-block:: console
+
+  kayobe overcloud host configure -t dnf
+
+StackHPC DOCA kernel modules will require the latest kernel version available in Ark for
+the current Rocky minor version. You should ensure that packages are up to date by running
+a package update, which can also be limited to hosts in the ``mlnx`` group.
+
+.. code-block:: console
+
+  kayobe overcloud host package update --packages "*" --limit mlnx
```

Comment:
Should probably also remind people here to reboot (and do they also need to run the reset bls playbook?) and also maybe a suggestion to do rolling reboots otherwise it could be very disruptive